File: pset6/credit/credit.py
def main():
    while True:
        CCn = input("Type your credit card number: ")
        if CCn.isalpha() == False:
            if int(CCn) >= 0:
                break
   
    CCn = int(CCn)

    Sumf = CheckSum(CCn)
    
    if Sumf == 0:
        CCcheck(CCn)
    else:
        print("INVALID")

def CCcheck(CCn):
    if (CCn >= 340000000000000 and CCn < 350000000000000) or (CCn >= 370000000000000 and CCn < 380000000000000):
        print("AMEX")
    elif CCn >= 5100000000000000 and CCn < 5600000000000000:
        print("MASTERCARD")
    elif (CCn >= 4000000000000 and CCn < 5000000000000) or (CCn >= 4000000000000000 and CCn < 5000000000000000):
        print("VISA")
    else:
        print("INVALID 1")
        return 0

def CheckSum(CCn):
    n = 1; dig = 1; sumtraco = 0; sumntraco = 0; count = 2

    while CCn // n > 0:
        if count % 2 != 0:
            dig = (CCn // n) % 10
            dig *= 2
            if dig >= 10:
                dig = (dig % 10) + (dig // 10)
            sumtraco += dig
        else:
            dig = (CCn // n) % 10
            sumntraco += dig
        count += 1
        n *= 10

    Sum = sumntraco + sumtraco

    Sumf = Sum % 10

    print(Sum)
    return Sumf

main()

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int checksum(long ccn);
int cccheck(long ccn);


int main()
{
    long ccn;
    do
    {
        ccn = get_long("Type your credit card number ");
    }
    while (ccn <= 0);

    int sumf = checksum(ccn);
    if (sumf == 0)
        cccheck(ccn);
    else
        printf("INVALID\n");
}

int cccheck(long ccn)
{
    if ((ccn >= 340000000000000 && ccn < 350000000000000) || (ccn >= 370000000000000 && ccn < 380000000000000))
        printf("AMEX\n");
    else if (ccn >= 5100000000000000 && ccn < 5600000000000000)
        printf("MASTERCARD\n");
    else if ((ccn >= 4000000000000 && ccn < 5000000000000) || (ccn >= 4000000000000000 && ccn < 5000000000000000))
        printf("VISA\n");
    else
        printf("INVALID\n");
        return 0;
}





int checksum(long ccn)
{
    long n=1;
    int dig = 1;
    int sumtraco = 0;
    int sumntraco = 0;
    
    for(int count = 2;ccn / n > 0 ; n = n*10)
    {
        if (count % 2 != 0)
        {
            dig = (ccn / n) % 10;
            dig = dig * 2;
            if (dig >= 10)
                dig = (dig % 10) + (dig / 10);
            sumtraco = sumtraco + dig;
        }
        else
        {
            dig = (ccn / n) % 10;
            sumntraco = sumntraco + dig;
        }
        count++;
    }
    int sum = sumntraco + sumtraco;
    
    int sumf = sum % 10;
    
    printf("%d\n",sum);
    return sumf;
}


